sdk/src/main/java/com/eegeo/mapapi/widgets/RouteViewHelper.java

```diff
@@ -0,0 +1,159 @@
+        final double epsilonSq = 1e-6;
+        float[] results = new float[1];
+        Location.distanceBetween(firstLocation.latitude, firstLocation.longitude, secondLocation.latitude, secondLocation.longitude, results);
+        return results[0] <= epsilonSq;
+    }
+
+    public static boolean areCoordinateElevationPairApproximatelyEqual(Pair<LatLng, Double> a, Pair<LatLng, Double> b) {
+        final double elevationEpsilon = 1e-3;
+
+        if(!areApproximatelyEqual(a.first, b.first)) {
+            return false;
+        }
+
+        return Math.abs(a.second - b.second) < elevationEpsilon;
+    }
+
+    public static void removeCoincidentPoints(List<LatLng> coordinates) {
+        if(coordinates == null) {
+            return;
+        }
+
+        Iterator<LatLng> it = coordinates.iterator();
+        LatLng prev = null;
+        LatLng current = it.next();
+        while (it.hasNext()) {
+            prev = current;
+            current = it.next();
+            if (areApproximatelyEqual(current, prev)) {
+                it.remove();
+            }
+        }
+    }
+
+    public static void removeCoincidentPointsWithElevations(List<LatLng> coordinates, List<Double> perPointElevations) {
+        List<Pair<LatLng, Double>> pairsList = new ArrayList<>(coordinates.size());
+
+        for(int i=0; i<coordinates.size(); i++) {
+            LatLng coordinate = coordinates.get(i);
+            Double elevation = perPointElevations.get(i);
+
+            pairsList.add(new Pair<>(coordinate, elevation));
+        }
+
+        int i = 1;
+        while (i < pairsList.size()) {
+            Pair<LatLng, Double> prev = pairsList.get(i-1);
+            Pair<LatLng, Double> current = pairsList.get(i);
+
+            if (areCoordinateElevationPairApproximatelyEqual(current, prev)) {
+               pairsList.remove(i-1);
+               coordinates.remove(i-1);
+               perPointElevations.remove(i-1);
+            } else {
+                i++;
+            }
+        }
+    }
+
+    public static RoutingPolylineCreateParams  makeNavRoutingPolylineCreateParams(List<LatLng> coordinates, int color, String indoorId, int indoorFloorId) {
+        return new RoutingPolylineCreateParams(coordinates, color, indoorId, indoorFloorId, null);
+    }
+
+    public static RoutingPolylineCreateParams  makeNavRoutingPolylineCreateParams(List<LatLng> coordinates, int color, String indoorId, int indoorFloorId, float heightStart, float heightEnd) {
+        return new RoutingPolylineCreateParams(coordinates, color, indoorId, indoorFloorId, Arrays.asList(Double.valueOf(heightStart), Double.valueOf(heightEnd)));
+    }
+
+    public static List<RoutingPolylineCreateParams> createLinesForRouteDirection(RouteStep routeStep, int color) {
+        List<RoutingPolylineCreateParams> results = new ArrayList<>();
+
+        RouteViewHelper.removeCoincidentPoints(routeStep.path);
```

Comment:
Are we modifying the route step directly here? Meaning it will remove points from the path that will later be used by point on path queries. Expect that this can have side effects.

Reply:
done